locales/en/combine_data.md

```diff
@@ -0,0 +1,75 @@
+
+@@ code=combine_data/combine_data.03.js @@
+
+@@ code=combine_data/combine_data.03.out @@
+
+First, we loop over each `article`, where we take its `brand_id` to look up the corresponding `brand` using the native `filter` function. Note that this function returns an array and we expect it to have only one element. In case there is no corresponding `brand`, `result[0]` will be `undefined`, and in order to prevent an error (something like `result[0] is undefined`), we use the ternary operator.
+ 
+Also, as we no longer need `brand_id` after the lookup has been done, we can safely delete it. 
+
+If we want to join by more than one attribute, we can modify the filter function as we want, e.g.: 
+
+@@ code=combine_data/combine_data.03b.js @@
+
+### Using a generic and more efficient approach 
+
+A more generic, and also more performant version of a join is proposed below (abbreviated from [this Stackoverflow answer](http://stackoverflow.com/questions/17500312/is-there-some-way-i-can-join-the-contents-of-two-javascript-arrays-much-like-i/17500836#17500836)). Its output is equivalent to the one of the above method.
+
+@@ code=combine_data/combine_data.04.js @@
+
+Because above defined function creates an index for the `lookupTable` (in our case `brands`) in the first iteration, it runs considerably faster than the previously shown method. Also, via a callback, it allows us to directly define which keys (or "attributes") we want to retain in the resulting, joined array (`output`). It is used like so: 
+
+@@ code=combine_data/combine_data.05.js @@
+
+@@ code=combine_data/combine_data.05.out @@
+
+Note that we don't modify `articles` *in place* but create a new array. 
+
+## Add rows from different data sets
+
+Let's say we want to load a huge data set from the server, but because of network performance reasons, we load it in three chunks and reassemble it on the client side. Using Queue.js, as illustrated in [reading in data](read_data.html), we get the data and immediately combine it. For this, we can use D3's `d3.merge` to combine the single arrays one after another. In database terms, this operation is called "union".
+
+@@ code=combine_data/combine_data.06.js @@
+
+<div class="aside">This code is using d3.js</div>
+
+@@ code=combine_data/combine_data.06.out @@
+
+Note that the argument passed to `d3.merge` must be an array itself, which is why we use the square brackets. 
+
+## Add columns from different data sets
+
+In the last case, we have two or more data sets which were split *vertically*, in other words, attributes describing the same observations are stored in different data sets and need to be combined. This implies that all data sets have the same length. For example, `dataset_1` below contains two observations of attribute `a` and attribute `b`, while `dataset_2` contains the same two observations, but of attribute `c` and `d`. 
```

Comment:
Maybe make it a bit clearer that both objects reference to the same conceptual entity, so merging them makes sense because at every index of both datasets we'll have separate information about the same entity?